File: UsingCsharp/UsingCsharp/Lessons/GoogleInterview.cs
namespace UsingCsharp.Lessons;

public class GoogleInterview
{
    public static bool DoesContainMatch(int[] arrayOne, int[] arrayTwo)
    {
        var uniqueSet = new HashSet<int>();
        foreach (var number in arrayOne)
        {
            uniqueSet.Add(number);
        }

        return arrayTwo.Any(t => uniqueSet.Contains(t));
    }
}

File: UsingTypeScript/src/Lessons/GoogleInterview.ts
export class GoogleInterview {
    public static DoesContainMatch(arrayOne: number[], arrayTwo: number[]): boolean {
        const uniqueSet: Set<number> = new Set();

        for (const item of arrayOne) {
            uniqueSet.add(item);
        }

        for (const item of arrayTwo) {
            if (uniqueSet.has(item)) return true;
        }

        return false;
    }
}